mlflow/utils/autologging_utils/__init__.py

```diff
@@ -427,6 +429,23 @@ def autolog(*args, **kwargs):
             if name != "mlflow" and get_autologging_config(name, "disable", True):
                 return
 
+            # The `disable_for_unsupported_versions` is only effective when the flavor name is
+            # listed in `FLAVOR_TO_MODULE_NAME_AND_VERSION_INFO_KEY` mapping. The mapping should
+            # contain all the flavors support autologging, but may not be up-to-date. To avoid
```

Comment:
Is it challenging to ensure `FLAVOR_TO_MODULE_NAME_AND_VERSION_INFO_KEY` is up-to-date?

Reply:
Maybe we can construct similar mapping from ml_package_versions.py, with some tweak of adding module name? I can give it a try.